packages/openapi-fetch/test/index.test.ts

```diff
@@ -142,6 +144,11 @@ describe("client", () => {
       }
     });
 
+    it("provides a Client type", () => {
+      const client = createClient<paths>({ baseUrl });
+      expectTypeOf(client).toEqualTypeOf<Client<paths>>();
```

Comment:
These tests are great, thank you. I’d like to have more of these in general. And since these can run inside a `test-d.ts` test, I’ve been thinking it would be good to separate the type vs runtime tests (and have more of these granular type assertions).

What are your thoughts on splitting runtime vs type tests? And could these be good additions to get that rolling?

Reply:
This is an excellent question.

Libraries I've been working on so far (notably Scala.js) *had* to split runtime / type tests because of infrastructure restrictions (or more specifically test cases that should compile and test cases that shouldn't compile).

Personally, I often found that annoying. Because you end up with test cases like (pseudocode):

```
assert(max([1]) == 1)
assert(max([1,2]) == 2)
```

And somewhere completely different:

```
assertFailsCompile(max([]))
```

What we can do with TS is much nicer IMO:

```
assert(max([1]) == 1)
assert(max([1,2]) == 2)
assertThrows(
  // @ts-expect-error
  max([])
)
```

That all being said: IMO it does make sense to split tests by some axis (e.g. *the functionality* offered, independent of whether the functionality is offered by the type system or the runtime system or a combination thereof).

So in this case, it is true we are testing the `Client` type function and its inference (w/o inference issues, this test would not be necessary, one could simply read the return type of `createClient`). So this could / should indeed go into a different "describe" (and/or even a different file).

Now, along which axes to split tests (exported elements, "features", ...) is not easy. However, it is also something that can be done very ad-hoc, so there is no need to "get it right" the first time.

Splitting what is in `index.test.js` into multiple files is IMO definitely a good idea (merely based on the LOC count of `index.test.js`).

In summary:
- Splitting tests into different categories / units / ... is IMO a good thing.
- I'm not sure typing-time / run-time is the best splitting axis though.

Reply:
> In summary:
>
> - Splitting tests into different categories / units / ... is IMO a good thing.
> - I'm not sure typing-time / run-time is the best splitting axis though.

You’ve convinced me 🙂. Let’s split up the tests into more meaningful divisions then. I think the test suite “feels” complete currently because it’s just too big a file. But in reality is missing several cases still.

Can be a followup obviously! Thanks for weighing in.